### PR TITLE
Fix memory leak in NotNode

### PR DESCRIPTION
**Pull request by uklotzde**
A bigger C++11 refactoring for master will follow... ;)

### CODE REVIEW

File: src/library/searchquery.h [hunk 1]
```diff
@@ -57,6 +57,7 @@ class AndNode : public GroupNode {
 class NotNode : public QueryNode {
   public:
     explicit NotNode(QueryNode* pNode);
+    ~NotNode();
```
Comment (ywwg) [inline src/library/searchquery.h]: destructors should be virtual

File: src/library/searchquery.cpp [hunk 2]
```diff
@@ -107,6 +107,10 @@ NotNode::NotNode(QueryNode* pNode)
         : m_pNode(pNode) {
 }
 
+NotNode::~NotNode() {
+    delete m_pNode;
```
Comment (ywwg) [inline src/library/searchquery.cpp]: m_pnode is passed in through the constructor, does this object own that pointer?  If not, it shouldn't delete it.

File: src/library/searchquery.h (hunk 1 above)
Comment (uklotzde) [inline src/library/searchquery.h]: Only the destructor in the base class needs to be virtual. The keyword 'virtual' is redundant in derived classes. For functions we will use 'override' in derived classes (C++11), but there is nothing similar for the destructor.

File: src/library/searchquery.cpp (hunk 2 above)
Comment (uklotzde) [inline src/library/searchquery.cpp]: Both the GroupNode and the NotNode own the pointers of their component(s). But only the GroupNode deletes them in the destructor.

File: src/library/searchquery.h (hunk 1 above)
Comment (uklotzde) [inline src/library/searchquery.h]: The refactoring for master will render all destructors of derived classes obsolete, btw ;)

File: src/library/searchquery.cpp (hunk 2 above)
Comment (uklotzde) [inline src/library/searchquery.cpp]: This question becomes obsolete with the refactoring for master which consequently uses std::unique_ptr. Another nice side effect of C++11 :)

Comment (kain88-de) [inline src/library/searchquery.cpp]: :+1: for the usage of `std::unique_ptr`. 

Do you mind splitting your C++11 refactor into several PR's. We use raw pointers a lot in mixxx and several medium sized PR's are just easier to review.

**Comment by daschuer** [comment]
> A bigger C++11 refactoring for master will follow... ;)

We have to be careful not to fail our SoundSource release one or two month after 1.12.0 release. 
So I would prefer to delay any mass refactoring without a bug or a feature improvement after the SoundSource release.   
How about setting up a c++11 refactoring branch, where we can collect all refectoring PRs in small steps, instead of piling up on single PR.

**Comment by uklotzde** [comment]
The refactoring for master iswill actually just a cleanup, removing of redundant code and important comments, because the existing code has some design issues (like calling a virtual function from a constructor).

**Comment by uklotzde** [comment]
Ups, accidentally closed the PR.

**Comment by rryan** [comment]
> We have to be careful not to fail our SoundSource release one or two month after 1.12.0 release. 
> So I would prefer to delay any mass refactoring without a bug or a feature improvement after the SoundSource release.

Code that's ready should merge to master. 

However, going back to my original warning about C++11 support -- please don't refactor things just because we have new toys available. i.e. don't convert all QScopedPointer usage to unique_ptr "just because".

**Pull request merged**